### PR TITLE
add level and ls command to log

### PR DESCRIPTION
**Pull request by frrist**
This PR add commands:
```
  go-filecoin log level <subsystem> <level> - Change the logging level.
  go-filecoin log ls                        - List the logging subsystems.
```
Prior art here: https://github.com/ipfs/go-ipfs/blob/master/core/commands/log.go

### CODE REVIEW

File: commands/log.go [hunk 1]
```diff
@@ -15,7 +27,9 @@ output of a running daemon.
 	},
 
 	Subcommands: map[string]*cmds.Command{
-		"tail": logTailCmd,
+		"tail":  logTailCmd,
+		"level": logLevelCmd,
```
Comment (phritz) [inline commands/log.go]: alpha

File: commands/log.go [hunk 2]
```diff
@@ -32,3 +46,69 @@ Outputs event log messages (not other log messages) as they are generated.
...
+	Helptext: cmdkit.HelpText{
+		Tagline: "Change the logging level.",
+		ShortDescription: `
+Change the verbosity of one or all subsystems log output. This does not affect
+the event log.
+`,
+	},
+
+	Arguments: []cmdkit.Argument{
+		// TODO use a different keyword for 'all' because all can theoretically
```
Comment (phritz) [inline commands/log.go]: my suggestion is remove this todo. if it does clash then someone can fix it. or if you want give the level first and then an optional subsystem second on the command line. if they leave the subsystem off then treat that as all. we definitely definitely should not try to use * on the command line.

File: commands/log.go [hunk 3]
```diff
@@ -32,3 +46,69 @@ Outputs event log messages (not other log messages) as they are generated.
...
+			One of: debug, info, warning, error, critical.
+		`),
+	},
+
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		args := req.Arguments
+		subsystem, level := args[0], args[1]
+
+		if subsystem == logAllKeyword {
+			subsystem = "*"
```
Comment (phritz) [inline commands/log.go]: if per comment at top of file args 0 is an array of all files then how does this assignment work?

File: commands/log.go (hunk 2 above)
Comment (frrist) [reply commands/log.go]: I like the idea of having subsystem be optional, will go with that

File: commands/log.go (hunk 3 above)
Comment (frrist) [reply commands/log.go]: yup, this is gone if we make subsystem an option

**Pull request merged**